core/state/trie_prefetcher.go

```diff
@@ -251,42 +198,30 @@ func newSubfetcher(db Database, state common.Hash, owner common.Hash, root commo
 func (sf *subfetcher) schedule(keys [][]byte) {
 	// Append the tasks to the current queue
 	sf.lock.Lock()
+
 	sf.tasks = append(sf.tasks, keys...)
 	sf.lock.Unlock()
-
 	// Notify the prefetcher, it's fine if it's already terminated
 	select {
-	case sf.wake <- struct{}{}:
```

Comment:
It might be possible that the prefetching loop is already terminated, but a new task is scheduled.

It doesn't satisfy our requirement, isn't it? We should somehow return a flag saying: the given task won't be executed as the prefetcher is already terminated?

Reply:
Afaict, we will only ever call `triePrefetcher.prefetch` before `triePrefetcher.close` has been called.  and they are not invoked concurrently.  so this case you bring up shouldn't happen (?).

Reply:
right